chainermn/iterators/multi_node_iterator.py

```diff
@@ -29,28 +41,38 @@ def __init__(self, actual_iterator, communicator, rank_master):
     def __next__(self):
         try:
             batch = self.actual_iterator.__next__()
-            stop = False
+            first_elem = batch[0]
+            valid_data_type = _is_valid_type(first_elem)
             is_paired_dataset = isinstance(batch, list) \
-                and isinstance(batch[0], tuple) and len(batch[0]) == 2
+                and isinstance(first_elem, tuple) and len(first_elem) == 2
+            stop = False
         except StopIteration:
             stop = True
+            valid_data_type = False
             is_paired_dataset = False
 
         is_new_epoch = self.actual_iterator.is_new_epoch
 
         # Notify the followings to slave iterators:
         # 1. whether stop signal is received before broadcasting data.
-        # 2. whether dataset is paired.
-        # 3. is_new_epoch.
-        # 4. current_position.
-        _info = numpy.ones((4, )) \
-            * [int(stop), int(is_paired_dataset),
+        # 2. whether type of batch element is valid.
+        # 3. whether dataset is paired.
+        # 4. is_new_epoch.
+        # 5. current_position.
+        _info = numpy.ones((5, )) \
```

Comment:
Given that this is a refactoring patch, I'd want this conversion as a symmetric functions for tests. Maybe we could increase the number of elements in future. Let's have unit tests for this conversions for future. Like this:

```python
def _build_ctrl_msg(stop, is_valid_data_type, is_paired_dataset, is_new_epoch, current_pos):
     ...

def _parse_ctrl_msg(msg):
    ....

def test_msg_conversion():
    msg = [....]
    assert msg == _build_ctr_msg(_parse_ctrl_msg(msg))
```